File: C++/quick_sort.cpp
#include<bits/stdc++.h>
using namespace std;
int par(int a[],int s,int e)
{
	int pivot = a[s];
	int pidx = s;
	int i = s;
	for(int j=s+1;j<=e;j++)
	{
		if(a[j]<=pivot)
		{
			i++;
			swap(a[i],a[j]);
		}
	}
	swap(a[i],a[pidx]);
	return i;
}
int quick_sort(int a[], int start, int end)
{	int idx;
	if(start < end)
	{
		idx = par(a,start,end);
		
		quick_sort(a,start,idx - 1);
		quick_sort(a,idx + 1, end);
	}
	
}
int main()
{
	int a[] = {12,4,5,15,6};
	cout<<"The array before sorting is: ";
	for(int i=0;i<5;i++)
	{
		cout<<a[i]<<" ";
	}
	cout<<"\n";
	quick_sort(a,0,4);
	cout<<"The array after sorting is: ";
	for(int i=0;i<5;i++)
	{
		cout<<a[i]<<" ";
	}
	return 0;
}

File: Java/QuickSort.java
public class QuickSort {
    static void swap(int [] arr, int index1, int index2){
        int temp = arr[index1];
        arr[index1] = arr[index2];
        arr[index2] = temp;
    }

    static int partition(int [] arr, int left, int right) {
        int pivot = arr[right];
        int i = (left-1);
        for (int j=left; j<right; j++) {
            if (arr[j] <= pivot) {
                i++;
                swap(arr, i, j);
            }
        }
        swap(arr, i+1, right);
        return i+1;
    }

    static void sort(int [] arr, int left, int right) {
        if (left < right) {
            int pivot = partition(arr, left, right);
            sort(arr, left, pivot-1);
            sort(arr, pivot+1, right);
        }
    }

    public static void main(String[] args) {
        // print
        int [] arr = {10, 7, 8, 9, 1, 5};
        System.out.println("before quick sort");
        for(int i=0; i<arr.length; i++){
            System.out.print(arr[i] + " ");
        }
        System.out.println("");

        // quick sort
        QuickSort.sort(arr, 0, arr.length-1);

        // print
        System.out.println("quick sort!");
        for(int i=0; i<arr.length; i++){
            System.out.print(arr[i] + " ");
        }
    }
} 